server/sockets.js

```diff
@@ -96,6 +97,25 @@ function socketConnection(socket) {
 			return;
 		}
 
+		if (message.data.type === "doc") {
+			getBoard(boardName).then(boardData => {
+				let existingDocuments = 0;
+				for (key in boardData.board) {
+					if (boardData.board[key].type === "doc") {
+						existingDocuments += 1;
+					}
+					if (existingDocuments >= config.MAX_DOCUMENTS) {
+						console.warn("Received too many documents");
+						return;
+					}
+				}
```

Comment:
The number of attached documents should probably be kept in the `BoardData` object (in `boardData.js`), and not recomputed from scratch on every document upload.

Reply:
I have been hesitant to change the way the data is stored so far but I see the need for some metadata storage. This would be useful for e.g. the preview and export functionality where we could set `lastmodified` and `lastgenerated` attributes to prevent us from generating the svg anew every time.

How about modifying the structure as follows?
```
{
   docCount: ...
   data: ...
}
```

Reply:
I'd like to avoid breaking backwards compatibility by changing the file format. We may come to it at some point, but we'll need to take the time to discuss the design in depth before. What I was suggesting above was to keep the document count in memory in the `BoardData` object. The count will be computed on disk load, and then incremented on document addition.

Reply:
That is a good idea that would also work for my other scenario.